thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -175,6 +175,10 @@ private void storeData(AnomalyDetectionOutputContext anomalyDetectionOutputConte
 
     DataFilter dataFilter = DataFilterFactory.fromSpec(anomalyFunctionSpec.getDataFilter());
     for (DimensionMap dimensionMap : anomalyDetectionInputContext.getDimensionMapMetricTimeSeriesMap().keySet()) {
+      if (Thread.interrupted()) {
```

Comment:
How the interrupt is triggered?

Reply:
With interruption, what will be the status of the task and job? If it is ongoing or failed, what will be the plan to recover that after the system is up?

Reply:
Interrupt happens at process level, e.g. via SIGINT, and is passed down to the threads via the ExecutorService's shutdownNow(). The exit state of the task should be "FAILED" and recover by the same means you recover from other errors during detection.

Reply:
Is it possible to apply to interrupt a specific detection job? FYI, currently the only way to recover the failed jobs is triggering replay offline.

Reply:
Also, this works for the detection runs normally. How do we handle when there is a zombie task? For example, the detection stops when running on one of the dimensions. It won't enter into the if statement.

Reply:
Can you add a task and job status as INTERRUPTED, so that we know this task/job was interrupted but failed?

Reply:
Detection jobs a currently a single-task monolith, so interruption is all or nothing. Zombie tasks stuck inside an algorithm is out of scope - this has to be addressed by the developer of the algorithm (but can be done in the exact same way)